controllers/mirroring/mirroring_controller.go

```diff
@@ -0,0 +1,448 @@
+
+	peerClientIDs := []string{}
+	nameToStorageConsumer := map[string]*ocsv1alpha1.StorageConsumer{}
+	for localClientID, peerClientID := range r.clientMappingConfig.Data {
+		// for internal mode, we need only blockPool mirroring, hence skipping this for the special key "internal"
+		if localClientID == internalKey {
+			continue
+		}
+		// Check if the storageConsumer with the ClientID exists
+		storageConsumers := &ocsv1alpha1.StorageConsumerList{}
+		if err := r.list(storageConsumers, client.MatchingFields{"clientID": localClientID}); err != nil {
+			return ctrl.Result{}, err
+		}
+		if len(storageConsumers.Items) != 1 {
+			return ctrl.Result{}, fmt.Errorf("expected 1 StorageConsumer but got %v", len(storageConsumers.Items))
+		}
+		nameToStorageConsumer[storageConsumers.Items[0].Name] = &storageConsumers.Items[0]
+		peerClientIDs = append(peerClientIDs, peerClientID)
+	}
+
+	response, err := ocsClient.GetStorageClientsInfo(r.ctx, r.storageClusterPeer.Status.PeerInfo.StorageClusterUid, peerClientIDs)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	peerClientIDToRadosNamespace := map[string]string{}
+	for i := range response.ClientsInfo {
+		clientInfo := response.ClientsInfo[i]
+		peerClientIDToRadosNamespace[clientInfo.ClientID] = clientInfo.RadosNamespace
+	}
+
+	radosNamespaceList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+	if err = r.list(radosNamespaceList, client.InNamespace(r.storageClusterPeer.Namespace)); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to list CephBlockPoolRadosNamespace: %w", err)
+	}
+
+	for i := range radosNamespaceList.Items {
+		rns := &radosNamespaceList.Items[i]
+		consumer := nameToStorageConsumer[rns.GetLabels()[controllers.StorageConsumerNameLabel]]
+		if consumer == nil {
+			continue
+		}
+		clientID := consumer.Status.Client.ID
+		radosNamespace := peerClientIDToRadosNamespace[r.clientMappingConfig.Data[clientID]]
+		if radosNamespace == "" {
+			continue
+		}
+		rns.Spec.Mirroring = &rookCephv1.RadosNamespaceMirroring{
+			RemoteNamespace: ptr.To(radosNamespace),
+			Mode:            "image",
+		}
+		if err := r.update(rns); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update radosnamespace %s", rns.Name)
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *MirroringReconciler) get(obj client.Object) error {
+	return r.Client.Get(r.ctx, client.ObjectKeyFromObject(obj), obj)
+}
+
+func (r *MirroringReconciler) list(obj client.ObjectList, listOptions ...client.ListOption) error {
+	return r.Client.List(r.ctx, obj, listOptions...)
+}
+
+func (r *MirroringReconciler) update(obj client.Object, opts ...client.UpdateOption) error {
+	return r.Client.Update(r.ctx, obj, opts...)
+}
+
+func (r *MirroringReconciler) delete(obj client.Object, opts ...client.DeleteOption) error {
+	return r.Client.Delete(r.ctx, obj, opts...)
+}
+
+func (r *MirroringReconciler) own(obj client.Object) error {
+	return controllerutil.SetControllerReference(r.storageClusterPeer, obj, r.Scheme)
+}
```

Comment:
An r.Own util is created to have an object owned by the input CR of this controller. This is not the case here. And this controller should not mark ownership for another controller's CR